### PR TITLE
add support for phi-2 model

### PR DESCRIPTION
**Pull request by salykovaa**
Following PR adds basic Phi-2 support
1) add phi2 model config
2) test phi2 model against hf reference model
3) update phi tutorial to include phi2
4) update chat/base.py prompt format for phi2

Steps to inference the phi-2 model:
1. ```python scripts/download.py --repo_id microsoft/phi-2 --from_safetensors True```
2. ```python scripts/convert_hf_checkpoint.py --checkpoint_dir checkpoints/microsoft/phi-2```
3. ```python chat/base.py --checkpoint_dir checkpoints/microsoft/phi-2/```
```
>> Prompt: Write a detailed analogy between mathematics and a lighthouse.
>> Output: Mathematics is like a lighthouse. Mathematics provides a method to guide us through the sometimes chaotic and confusing waters of life. It provides a structured approach to problems which can help us find our way and provide direction. Just as a lighthouse keeps watch over the sea, mathematics can provide us with the tools to try and make sense of the world. Furthermore, just as a lighthouse keeps a watchful eye on the horizon, mathematics can help us reach our goals by showing us the way.
```

### CODE REVIEW

File: lit_gpt/config.py [hunk 1]
```diff
@@ -1148,7 +1148,20 @@ def norm_class(self) -> Type:
         shared_attention_norm=True,
         lm_head_bias=True,
         gelu_approximate="tanh",
-    )
+    ),
+     dict(
```
Comment (carmocca) [inline lit_gpt/config.py:1152]:
```suggestion
    # https://huggingface.co/microsoft/phi-2/blob/main/config.json
    dict(
```

Also looks like the dict is indented one extra space

**Review by carmocca — COMMENTED** [review]
Can you also update https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/download_phi15.md? And rename the file. It's also referenced in the README table.

And create a copy of the phi 1.5 test: https://github.com/Lightning-AI/lit-gpt/blob/main/tests/test_model.py#L273

**Comment by salykovaa** [comment]
hi @carmocca, could you please review the code? I've updated the phi tutorial and tests

**Review by carmocca — APPROVED** [review]
Really good job @salykovaa!

**Pull request merged**